gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/common/RedirectUriValidationHandler.java

```diff
@@ -44,25 +52,58 @@ public class RedirectUriValidationHandler implements Handler<RoutingContext> {
 
     private final Domain domain;
     private final RedirectUriValidator redirectUriValidator;
+    private Function<String, Maybe<JWT>> tokenVerifier;
 
     public RedirectUriValidationHandler(Domain domain) {
         this.domain = domain;
         this.redirectUriValidator = new RedirectUriValidator();
+        this.tokenVerifier = t -> Maybe.empty();
+    }
+
+    public RedirectUriValidationHandler(Domain domain, UserService userService) {
+        this.domain = domain;
+        this.redirectUriValidator = new RedirectUriValidator();
+        this.tokenVerifier = t -> userService.verifyToken(t)
+                .map(UserToken::getToken);
     }
 
     @Override
     public void handle(RoutingContext context) {
         final Client client = context.get(ConstantKeys.CLIENT_CONTEXT_KEY);
+        getOperation(context)
+                .subscribe(operation -> {
+                            // proceed redirect_uri parameter
+                            parseRedirectUriParameter(context, client, operation);
```

Comment:
**thought**: Are you sure it will work in case of failure ? I would move this method call into a `map` operator to simply manage the context::next or context::fail into the subscription. In the current implementation, my fear is if the redirect url is not valid, vertx will never be inform about the failure :thinking: maybe I'm wrong

Reply:
you're right, that's exactly what caused some tests to fail